vars/notifyBuildFixed.groovy

```diff
@@ -35,7 +35,9 @@ def call(Map args = [:]) {
   }
 
   try {
-    if (currentBuild.getPreviousBuild()?.getResult() == 'FAILURE') {
+    // Check if currentBuild has the getPreviousBuild method (may not exist in test environments)
+    def hasPreviousBuild = currentBuild?.hasProperty('getPreviousBuild') || currentBuild?.metaClass?.respondsTo(currentBuild, 'getPreviousBuild')
```

Comment:
Added checks for null and method existence to prevent runtime errors when Jenkins objects don't have expected properties or methods like getPreviousBuild.